generic-avro-reflect/src/main/java/org/apache/avro/reflect/Reflect2Data.java

```diff
@@ -171,6 +171,10 @@ private Type resolveType(final List<TypedValueAccessor> accessors, final Object
         final TypeToken<?> fieldToken = tt.resolveType(field.getGenericType());
         final TypedValueAccessor typedValueAccessor = new TypedValueAccessor(accessor);
 
+        if (fieldToken.getRawType().equals(Object.class) && !fieldToken.getType().getTypeName()
```

Comment:
When is the first case actually happening (equals Object)? I thought at that point, we are sure that we have some generic signature (see line 166).
Btw we should really start comment the cases ;)

Reply:
Actually,  we do have a generic signature here. However, we have to differentiate between nested structures and plain objects. 
The problem is that for classes with more than one generic we iterate for each generic through all fields (see line 155). Thus, for the second generic, the condition in line 178 does not match and we end up checking for subtypes in line 188. In the case of simple classes, we do not want to do that, while it is necessary for nested classes.